deployment/ccip/changeset/cs_chain_contracts.go

```diff
@@ -167,6 +307,143 @@ func UpdateOnRampsDests(e deployment.Environment, cfg UpdateOnRampDestsConfig) (
 	}}, nil
 }
 
+type UpdateFeeQuoterPricesConfig struct {
+	InitialPrices map[uint64]FeeQuoterPriceUpdatePerSource
+	MCMS          *MCMSConfig
+}
+
+type FeeQuoterPriceUpdatePerSource struct {
+	TokenPrices map[common.Address]*big.Int // token address -> price
+	GasPrices   map[uint64]*big.Int         // dest chain -> gas price
+}
+
+func (cfg UpdateFeeQuoterPricesConfig) Validate(e deployment.Environment) error {
+	state, err := LoadOnchainState(e)
+	if err != nil {
+		return err
+	}
+	for chainSel, initialPrice := range cfg.InitialPrices {
+		if err := deployment.IsValidChainSelector(chainSel); err != nil {
+			return fmt.Errorf("invalid chain selector: %w", err)
+		}
+		chainState, ok := state.Chains[chainSel]
+		if !ok {
+			return fmt.Errorf("chain %d not found in onchain state", chainSel)
+		}
+		if chainState.FeeQuoter == nil {
+			return fmt.Errorf("missing fee quoter for chain %d", chainSel)
+		}
+		if err := commoncs.ValidateOwnership(e.GetContext(), cfg.MCMS != nil, e.Chains[chainSel].DeployerKey.From, chainState.Timelock.Address(), chainState.FeeQuoter); err != nil {
```

Comment:
I think we'd also need to check that the timelock or deployerkey is enabled as a price updater